### PR TITLE
Fix the backport of #4512 into release-2.2 branch

### LINKED ISSUE #4511: AWSCluster AdditionalTags are not propagated to AWSMachine storage volumes
## Bug Description

The AWSCluster resources has a spec.AdditionalTags k-v map that is a place end users (aka me!) can place kvs that I want to use as resource tags in AWS (mostly for cost tracking). 

The idea is that these tags are propagated to all resources descendent from the AWSCluster, such as ec2 instances and their root storage volumes.

Today, these tags make it to ec2 instances but not the root storage volumes. Only the awsmachine.spec.AdditionalTags make it to the root storage volumes. The layering of AdditionalTags is great for providing overrides (or additions) at various levels but the `AdditionalTags` from the `AWSCluster` should propagate to all resources. 

I believe this is a bug; but I could imagine someone arguing that it is a design choice - it would be a surprising one and one that creates more work for the end user (as the user would need to explicitly also set AdditionalTags on the AWSMachine manifest).

## Fix

This is an easy fix. See PR shortly. 

Here's the `ensureStorageTags` method which only pulls tags from the machine L1121.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/b90c18cf1bac7308ace428245945a40e3099d758/controllers/awsmachine_controller.go#L1114-L1140

Just above its callsite, you can see that for ec2 instances the `ensureTags` method merges tags from the machine with those from the cluster through `machineScope.AdditionalTags()`.

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/b90c18cf1bac7308ace428245945a40e3099d758/controllers/awsmachine_controller.go#L585-L593

## Prior work

The `ensureStorageTags` method was introduced in this PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2463

### PR DESCRIPTION
**Pull request by cnmcavoy**
**What type of PR is this?**
/kind regression

**What this PR does / why we need it**: Regression from main branch, backport of #4512 into release-2.2 failed and introduced errors and missed adding a unit test. See conversation on slack by @philjb : https://kubernetes.slack.com/archives/CD6U2V71N/p1697658532510839 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4511 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Fixed: Additional Tags specified on the AWSCluster will now propagate to root storage volumes.
```

### CODE REVIEW

**Comment by philjb** [comment]
@richardcase @cnmcavoy  - how does this work now? Is there a new release of 2.2? aka 2.2.5?

Thanks for making this fix - I'd been meaning to do it. I just yesterday double checked the behavior of `main` and the 2.2.4 release. 2.2.4 with only the partial cherry pick is broken still with regards to #4512. The code in this PR should fix pushing tags to storage volumes on the 2.2 branch line. 

Same thing said differently, based on my (local) testing yesterday, #4512 should not be added to 2.2.4 release notes as the issue isn't fixed there. This pr should fix it in the next release of 2.2.

**Review by philjb — APPROVED** [review]
looks good to me. 

I tested (tilt based test setup) that someone with a cluster with CAPA 2.2.4 can upgrade to this patch (future 2.2.5) and it works properly. I was concerned that the prevAnnotation annotation might not push tag updates through after upgrades but it does. 

EDIT: for more context, `prevAnnotations` isn't even used when updating tags (which is good - prevents issues with an annotations state being wrong - but it doesn't mean more API tag requests - on every reconcile i believe); `prevAnnotations` is used when deciding when to update the annotation which says which tags were last applied.

**Comment by richardcase** [comment]
@philjb - once this merges we can thinking about doing a v2.2.5 release.

**Comment by richardcase** [comment]
We just need some to review and provide the lgtm

**Review by AndiDog — COMMENTED** [review]
/lgtm

**Pull request merged**
